Crosscompilation from linux to windows fails

# Expected
I expected this code to create liboqs.dll in bin folder.

> This will generate a bin/liboqs.dll file in the build directory that you will need to copy to Windows. You should configure it to be available via the PATH environment variable.

[Cross compilation wiki (linux -> windows dll)](https://github.com/open-quantum-safe/liboqs/wiki/Platform-specific-notes-for-building-liboqs#cross-compiling)

# Replication
```bash
sudo apt install cmake gcc gcc-mingw-w64 ninja-build
git clone https://github.com/open-quantum-safe/liboqs
cd liboqs
mkdir build
cd build
cmake -GNinja -DCMAKE_TOOLCHAIN_FILE=../.CMake/toolchain_windows-amd64.cmake -DOQS_DIST_BUILD=ON -DBUILD_SHARED_LIBS=ON ..
ninja
```
# Result
```
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
gcc is already the newest version (4:11.2.0-1ubuntu1).
gcc-mingw-w64 is already the newest version (10.3.0-14ubuntu1+24.3).
ninja-build is already the newest version (1.10.1-1).
cmake is already the newest version (3.22.1-1ubuntu1.22.04.1).
0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.
Cloning into 'liboqs'...
remote: Enumerating objects: 32578, done.
remote: Counting objects: 100% (6868/6868), done.
remote: Compressing objects: 100% (1793/1793), done.
remote: Total 32578 (delta 5625), reused 5528 (delta 5057), pack-reused 25710
Receiving objects: 100% (32578/32578), 139.05 MiB | 2.10 MiB/s, done.
Resolving deltas: 100% (23686/23686), done.
-- The C compiler identification is GNU 10.0.0
-- The ASM compiler identification is GNU
-- Found assembler: /usr/bin/x86_64-w64-mingw32-gcc
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/x86_64-w64-mingw32-gcc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Performing Test CC_SUPPORTS_WA_NOEXECSTACK
-- Performing Test CC_SUPPORTS_WA_NOEXECSTACK - Failed
-- Performing Test LD_SUPPORTS_WL_Z_NOEXECSTACK
-- Performing Test LD_SUPPORTS_WL_Z_NOEXECSTACK - Failed
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Failed
-- Check if compiler accepts -pthread
-- Check if compiler accepts -pthread - yes
-- Found Threads: TRUE  
-- Alg enablement unchanged
-- Looking for getentropy
-- Looking for getentropy - not found
-- Looking for aligned_alloc
-- Looking for aligned_alloc - not found
-- Looking for posix_memalign
-- Looking for posix_memalign - not found
-- Looking for memalign
-- Looking for memalign - not found
-- Looking for explicit_bzero
-- Looking for explicit_bzero - not found
-- Looking for memset_s
-- Looking for memset_s - not found
-- Could NOT find Doxygen (missing: DOXYGEN_EXECUTABLE) 
-- Configuring done
-- Generating done
-- Build files have been written to: /home/fundan3/Programming/Echo/client/install_dependencies/liboqs/build
[382/572] Linking C shared library bin/liboqs-internal.dll
FAILED: bin/liboqs-internal.dll lib/liboqs-internal.dll.a 
: && /usr/bin/x86_64-w64-mingw32-gcc   -Wl,--stack,16777216 -shared -o bin/liboqs-internal.dll -Wl,--out-implib,lib/liboqs-internal.dll.a -Wl,--major-image-version,0,--minor-image-version,0 src/common/sha3/xkcp_low/CMakeFiles/xkcp_low_keccakp_1600_plain64.dir/KeccakP-1600/plain-64bits/KeccakP-1600-opt64.c.obj src/common/sha3/xkcp_low/CMakeFiles/xkcp_low_keccakp_1600times4_serial.dir/KeccakP-1600times4/serial/KeccakP-1600-times4-on1.c.obj src/common/CMakeFiles/internal.dir/aes/aes.c.obj src/common/CMakeFiles/internal.dir/aes/aes_c.c.obj src/common/CMakeFiles/internal.dir/aes/aes128_ni.c.obj src/common/CMakeFiles/internal.dir/aes/aes256_ni.c.obj src/common/CMakeFiles/internal.dir/sha2/sha2.c.obj src/common/CMakeFiles/internal.dir/sha2/sha2_c.c.obj src/common/CMakeFiles/internal.dir/sha3/xkcp_sha3.c.obj src/common/CMakeFiles/internal.dir/sha3/xkcp_sha3x4.c.obj src/common/CMakeFiles/internal.dir/common.c.obj src/common/CMakeFiles/internal.dir/rand/rand_nist.c.obj  -lkernel32 -luser32 -lgdi32 -lwinspool -lshell32 -lole32 -loleaut32 -luuid -lcomdlg32 -ladvapi32 && :
/usr/bin/x86_64-w64-mingw32-ld: src/common/CMakeFiles/internal.dir/sha3/xkcp_sha3.c.obj:xkcp_sha3.c:(.text$OQS_SHA3_sha3_256+0x46): undefined reference to `pthread_once'
/usr/bin/x86_64-w64-mingw32-ld: src/common/CMakeFiles/internal.dir/sha3/xkcp_sha3.c.obj:xkcp_sha3.c:(.text$OQS_SHA3_sha3_256_inc_init+0x32): undefined reference to `pthread_once'
/usr/bin/x86_64-w64-mingw32-ld: src/common/CMakeFiles/internal.dir/sha3/xkcp_sha3.c.obj:xkcp_sha3.c:(.text$OQS_SHA3_sha3_256_inc_ctx_reset+0x17): undefined reference to `pthread_once'
/usr/bin/x86_64-w64-mingw32-ld: src/common/CMakeFiles/internal.dir/sha3/xkcp_sha3.c.obj:xkcp_sha3.c:(.text$OQS_SHA3_sha3_384+0x46): undefined reference to `pthread_once'
/usr/bin/x86_64-w64-mingw32-ld: src/common/CMakeFiles/internal.dir/sha3/xkcp_sha3.c.obj:xkcp_sha3.c:(.text$OQS_SHA3_sha3_384_inc_init+0x32): undefined reference to `pthread_once'
/usr/bin/x86_64-w64-mingw32-ld: src/common/CMakeFiles/internal.dir/sha3/xkcp_sha3.c.obj:xkcp_sha3.c:(.text$OQS_SHA3_sha3_384_inc_ctx_reset+0x17): more undefined references to `pthread_once' follow
collect2: error: ld returned 1 exit status
[395/572] Building C object src/sig/falcon/CMakeFiles/falcon_512_avx2.dir/pqclean_falcon-512_avx2/keygen.c.obj
ninja: build stopped: subcommand failed.

```
# System and enviorment
**Latest liboqs commit**
**Every package is updated**
```
System:
  Kernel: 5.15.0-94-generic x86_64 bits: 64 compiler: gcc v: 11.4.0 Desktop: Cinnamon 6.0.4
    tk: GTK 3.24.33 wm: muffin vt: 7 dm: LightDM 1.30.0 Distro: Linux Mint 21.3 Virginia
    base: Ubuntu 22.04 jammy
Machine:
  Type: Laptop System: Dell product: G3 3590 v: N/A serial: <superuser required> Chassis: type: 10
    serial: <superuser required>
  Mobo: Dell model: 061RYD v: A00 serial: <superuser required> UEFI: Dell v: 1.13.0
    date: 06/13/2021
Battery:
  ID-1: BAT0 charge: 40.9 Wh (100.0%) condition: 40.9/51.0 Wh (80.1%) volts: 12.6 min: 11.4
    model: SMP DELL PN1VN9A type: Li-poly serial: <filter> status: Full
CPU:
  Info: 6-core model: Intel Core i7-9750H bits: 64 type: MT MCP smt: enabled arch: Coffee Lake
    rev: A cache: L1: 384 KiB L2: 1.5 MiB L3: 12 MiB
  Speed (MHz): avg: 809 high: 867 min/max: 800/4500 cores: 1: 800 2: 800 3: 800 4: 800 5: 800
    6: 867 7: 843 8: 800 9: 800 10: 800 11: 804 12: 800 bogomips: 62399
  Flags: avx avx2 ht lm nx pae sse sse2 sse3 sse4_1 sse4_2 ssse3 vmx
Graphics:
  Device-1: Intel CoffeeLake-H GT2 [UHD Graphics 630] vendor: Dell driver: i915 v: kernel ports:
    active: eDP-1 empty: DP-1,HDMI-A-1 bus-ID: 00:02.0 chip-ID: 8086:3e9b class-ID: 0300
  Device-2: NVIDIA TU116M [GeForce GTX 1660 Ti Mobile] vendor: Dell driver: nvidia v: 535.154.05
    pcie: speed: 8 GT/s lanes: 8 ports: active: none empty: HDMI-A-2 bus-ID: 01:00.0
    chip-ID: 10de:2191 class-ID: 0300
  Display: x11 server: X.Org v: 1.21.1.4 driver: X: loaded: modesetting,nvidia
    unloaded: fbdev,nouveau,vesa gpu: i915 display-ID: :0 screens: 1
  Screen-1: 0 s-res: 1920x1080 s-dpi: 98 s-size: 499x280mm (19.6x11.0") s-diag: 572mm (22.5")
  Monitor-1: eDP-1 model: BOE Display res: 1920x1080 hz: 60 dpi: 142 size: 344x194mm (13.5x7.6")
    diag: 395mm (15.5") modes: 1920x1080
  OpenGL: renderer: Mesa Intel UHD Graphics 630 (CFL GT2) v: 4.6 Mesa 23.0.4-0ubuntu1~22.04.1
    direct render: Yes
Audio:
  Device-1: Intel Cannon Lake PCH cAVS vendor: Dell driver: sof-audio-pci-intel-cnl
    bus-ID: 00:1f.3 chip-ID: 8086:a348 class-ID: 0401
  Device-2: NVIDIA TU116 High Definition Audio vendor: Dell driver: snd_hda_intel v: kernel
    pcie: speed: 8 GT/s lanes: 8 bus-ID: 01:00.1 chip-ID: 10de:1aeb class-ID: 0403
  Sound Server-1: ALSA v: k5.15.0-94-generic running: yes
  Sound Server-2: PulseAudio v: 15.99.1 running: yes
  Sound Server-3: PipeWire v: 0.3.48 running: yes
Network:
  Device-1: Intel Cannon Lake PCH CNVi WiFi driver: iwlwifi v: kernel bus-ID: 00:14.3
    chip-ID: 8086:a370 class-ID: 0280
  IF: wlo1 state: up mac: <filter>
  Device-2: Realtek RTL8111/8168/8411 PCI Express Gigabit Ethernet vendor: Dell driver: r8169
    v: kernel pcie: speed: 2.5 GT/s lanes: 1 port: 3000 bus-ID: 03:00.0 chip-ID: 10ec:8168
    class-ID: 0200
  IF: enp3s0 state: down mac: <filter>
Drives:
  Local Storage: total: 476.94 GiB used: 121.14 GiB (25.4%)
  ID-1: /dev/nvme0n1 vendor: Getrich model: 512G NVMe SSD size: 476.94 GiB speed: 31.6 Gb/s
    lanes: 4 type: SSD serial: <filter> rev: V0926A0 temp: 35.9 C scheme: GPT
Partition:
  ID-1: / size: 180.02 GiB used: 121.11 GiB (67.3%) fs: ext4 dev: /dev/nvme0n1p5
  ID-2: /boot/efi size: 96 MiB used: 31.1 MiB (32.4%) fs: vfat dev: /dev/nvme0n1p1
Swap:
  ID-1: swap-1 type: file size: 2 GiB used: 1.2 MiB (0.1%) priority: -2 file: /swapfile
USB:
  Hub-1: 1-0:1 info: Hi-speed hub with single TT ports: 16 rev: 2.0 speed: 480 Mb/s
    chip-ID: 1d6b:0002 class-ID: 0900
  Device-1: 1-1:3 info: USB Optical Mouse type: Mouse driver: hid-generic,usbhid interfaces: 1
    rev: 1.1 speed: 1.5 Mb/s power: 100mA chip-ID: 30fa:0400 class-ID: 0301
  Hub-2: 2-0:1 info: Super-speed hub ports: 8 rev: 3.1 speed: 10 Gb/s chip-ID: 1d6b:0003
    class-ID: 0900
  Hub-3: 3-0:1 info: Hi-speed hub with single TT ports: 2 rev: 2.0 speed: 480 Mb/s
    chip-ID: 1d6b:0002 class-ID: 0900
  Hub-4: 4-0:1 info: Super-speed hub ports: 4 rev: 3.1 speed: 10 Gb/s chip-ID: 1d6b:0003
    class-ID: 0900
Sensors:
  System Temperatures: cpu: 60.0 C pch: 49.0 C mobo: N/A
  Fan Speeds (RPM): N/A
Repos:
  Packages: apt: 2349
  No active apt repos in: /etc/apt/sources.list
  Active apt repos in: /etc/apt/sources.list.d/official-package-repositories.list
    1: deb http: //packages.linuxmint.com virginia main upstream import backport
    2: deb http: //archive.ubuntu.com/ubuntu jammy main restricted universe multiverse
    3: deb http: //archive.ubuntu.com/ubuntu jammy-updates main restricted universe multiverse
    4: deb http: //archive.ubuntu.com/ubuntu jammy-backports main restricted universe multiverse
    5: deb http: //security.ubuntu.com/ubuntu/ jammy-security main restricted universe multiverse
  Active apt repos in: /etc/apt/sources.list.d/librewolf.sources
    1: deb [arch=amd64] https: //deb.librewolf.net focal main
Info:
  Processes: 337 Uptime: 5h 37m wakeups: 64565 Memory: 15.41 GiB used: 2.82 GiB (18.3%)
  Init: systemd v: 249 runlevel: 5 Compilers: gcc: 11.4.0 alt: 11/12 Client: Cinnamon v: 6.0.4
  inxi: 3.3.13
```
Probably most of this information is unnecessary but who know.

## Additional
I am not very smart and I am grateful that you are spending your time on me.
Thanks.

--- a/.CMake/compiler_opts.cmake
+++ b/.CMake/compiler_opts.cmake
@@ -206,6 +206,7 @@ elseif(CMAKE_C_COMPILER_ID STREQUAL "MSVC")
 endif()
 
 if(MINGW OR MSYS OR CYGWIN)
+    set(OQS_USE_PTHREADS OFF)
     add_compile_options(-Wno-maybe-uninitialized)
     if(CMAKE_VERSION VERSION_GREATER_EQUAL "3.13.0")
         add_link_options(-Wl,--stack,16777216)

